### PR TITLE
Port commands to Drush 9

### PR DESCRIPTION
**Pull request by kimpepper**

### CODE REVIEW

**Comment by larowlan** [comment]
with the move to OO is there a chance to remove a lot of duplication here?
a lot of this was copy/paste from base drush commands, with inheritance, much of it might not be needed?

**Comment by kimpepper** [comment]
Yep sure. Was more worried about losing support in drush 9 beta5

**Comment by larowlan** [comment]
is it being dropped?

**Comment by kimpepper** [comment]
Yep

On 6 Oct. 2017 8:43 am, "Lee Rowlands" <notifications@github.com> wrote:

is it being dropped?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/previousnext/drush_cmi_tools/pull/21#issuecomment-334600615>,
or mute the thread

File: src/Commands/DrushCmiToolsCommands.php [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    'ignore-list' => NULL,
+  ]) {
+    $this->logger()->debug('Starting Exporting.');
+    // Do the actual config export operation
+    // Determine which target directory to use.
+    if ($target = $options['destination']) {
+      $destination_dir = $target;
+      // It is important to be able to specify a destination directory that
+      // does not exist yet, for exporting on remote systems.
+      $this->fileSystem->mkdir($destination_dir, NULL, TRUE);
```
Comment (acbramley) [inline src/Commands/DrushCmiToolsCommands.php]: This is throwing a warning when the directory exists `mkdir(): File exists FileSystem.php:237`

File: src/Commands/DrushCmiToolsCommands.php [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    }
+    else {
+      $this->logger()->error('You must provide a --destination option');
+      return NULL;
+    }
+    $patterns = [];
+    if (isset($ignoreList)) {
```
Comment (acbramley) [inline src/Commands/DrushCmiToolsCommands.php]: Need to set this or check `$options['ignore-list']` instead

**Comment by acbramley** [comment]
All other functionality is working as expected, just need a fix for the ignore list and mkdir warnings. One thing to note though is we'll either need to up the log level, or add --verbose to our commands since info logs aren't being displayed for me.

**Comment by acbramley** [comment]
Retest looking really good!
```
root@5554d90f07f8:/data# make export
./bin/drush -r /data/app cexy -y --destination=/data/config-export --ignore-list=/data/drush/config-ignore.yml
Differences of the active config to the export directory:

 Collection  Config           Operation
             shield.settings  create

 // The .yml files in your export directory (/data/config-export) will be deleted and replaced with the active config.:
 // yes.

 [success] Configuration successfully exported to /data/config-export.
 [notice] Removed /data/config-export/shield.settings.yml according to ignore list.
```

File: composer.json [hunk 3]
```diff
@@ -1,15 +1,42 @@
...
+    },
+    "repositories": {
+        "packages.drupal": {
+            "type": "composer",
+            "url": "https://packages.drupal.org/8"
+        }
+    },
+    "require": {
+        "drupal/core": "^8.4",
+        "drush/drush": "^9.0.0-beta6"
```
Comment (jibran) [inline composer.json:22]: Let's not add require. People have been using standalone drush bin with this extension i.e. https://github.com/previousnext/drush_cmi_tools/pull/19#issuecomment-323539649

Comment (kimpepper) [reply composer.json:22]: Currently you can't run this as standalone. Drush 9 requires this to be a module in order to use the service container.

**Comment by kimpepper** [comment]
This was merged into 8.x-2.x

**Pull request closed without merging**